src/coreComponents/constitutive/permeability/DisplacementDependentPermeability.hpp

```diff
@@ -0,0 +1,173 @@
+  {
+    compute( displacementJump,
+             m_permeability[k][q],
+             m_dPerm_dDisplacement[k][q] );
+  }
+
+private:
+
+  /// dPermeability_dDisplacement
+  arrayView3d< real64 > m_dPerm_dDisplacement;
+
+  /// Threshold of shear displacement
+  real64 m_shearDispThreshold;
+
+  /// Maximum permeability multiplier
+  real64 m_maxPermMultiplier;
+
+  /// Initial permeability tensor
+  arrayView3d< real64 > m_iniPermeability;
+
+};
+
+
+class DisplacementDependentPermeability : public PermeabilityBase
+{
+public:
+
+  DisplacementDependentPermeability( string const & name, Group * const parent );
+
+  std::unique_ptr< ConstitutiveBase > deliverClone( string const & name,
+                                                    Group * const parent ) const override;
+
+  virtual void allocateConstitutiveData( dataRepository::Group & parent,
+                                         localIndex const numConstitutivePointsPerParentIndex ) override;
+
+  static string catalogName() { return "DisplacementDependentPermeability"; }
+
+  virtual string getCatalogName() const override { return catalogName(); }
+
+  /// Type of kernel wrapper for in-kernel update
+  using KernelWrapper = DisplacementDependentPermeabilityUpdate;
+
+  /**
+   * @brief Create an update kernel wrapper.
+   * @return the wrapper
+   */
+  KernelWrapper createKernelWrapper() const
+  {
+    return KernelWrapper( m_permeability,
+                          m_dPerm_dPressure,
+                          m_dPerm_dDisplacement,
+                          m_shearDispThreshold,
+                          m_maxPermMultiplier,
+                          m_iniPermeability );
+  }
+
+  struct viewKeyStruct : public PermeabilityBase::viewKeyStruct
+  {
+    static constexpr char const * dPerm_dDisplacementString() { return "dPerm_dDisplacement"; }
+    static constexpr char const * shearDispThresholdString() { return "shearDispThreshold"; }
+    static constexpr char const * maxPermMultiplierString() { return "maxPermMultiplier"; }
+    static constexpr char const * iniPermeabilityString() { return "iniPermeability"; }
+  } viewKeys;
+
+private:
+
+  /// dPermeability_dStrain
+  arrayView3d< real64 > m_dPerm_dDisplacement;
+
+  /// Threshold of shear strain
+  real64 m_shearDispThreshold;
+
+  /// Maximum permeability multiplier
+  real64 m_maxPermMultiplier;
+
+  /// Initial permeability tensor
+  arrayView3d< real64 > m_iniPermeability;
+
+};
+
+
+GEOSX_HOST_DEVICE
+GEOSX_FORCE_INLINE
+void DisplacementDependentPermeabilityUpdate::compute( real64 const ( &displacementJump )[3],
+                                                       arraySlice1d< real64 > const & permeability,
+                                                       arraySlice1d< real64 > const & dPerm_dDisplacement  ) const
+{ 
+  real64 const shearDisp = std::max( abs(displacementJump[1]), abs(displacementJump[2]) )
```

Comment:
I suggest we do this with an L2-norm of the shear displacement instead of the max norm.  `shearMag = sqrt(u1^2+u2^2)`.  We can then consistently linearize as `dShearMag_dDisp = (1/shearMag)*{u1,u2}`.

Reply:
Good idea, please check the update